### ISSUE TITLE
Features Requests / Possible Bugs

### ISSUE THREAD

**Issue by allanice001** [miniCruzer/mini-key-server#5]
Not sure if this is intentional or not, but the validation seems borked

After activating the the token (specifying a machine and user), neither are used in the validation (check) steps, thus it is possible for user paul to "steal" user sam's token, and still have activated software. Additionally, the software can be installed on multiple machines (with different machine names) and still return as license ok

A quick way to resolve this might be to have the activations stored in the db, and verified agains an active user (by using a foreign key or many-many to the users table), and having the api/check validate it against this table

HTH

**Comment by allanice001** [comment]
Examples of the above
### User: sam
> $ curl localhost:5001/api/check -X GET -d token=3CQ08J7RFCVA6M5RCH5QL5UXH -d user=sam -d machine=ryujin
> {"result": "ok"}

### User: paul
> curl localhost:5001/api/check -X GET -d token=3CQ08J7RFCVA6M5RCH5QL5UXH -d user=paul -d machine=ryujin
> {"result": "ok"}

### Machine: peanut
> $ curl localhost:5001/api/check -X GET -d token=3CQ08J7RFCVA6M5RCH5QL5UXH -d user=paul -d machine=peanut
> {"result": "ok"}

### Machine: raisin
> $ curl localhost:5001/api/check -X GET -d token=3CQ08J7RFCVA6M5RCH5QL5UXH -d user=paul -d machine=raisin
> {"result": "ok"}

**Comment by miniCruzer** [comment]
Yes this, this is intentional. Anyone can rename their computer or user name on a machine. Machine name and username are only memo fields to help you with fraud detection to who is using your software and on what machine.

**Comment by miniCruzer** [comment]
You can achieve this with the hwid parameter. If supplied, the key can only be checked on that one machine. I will have to come back around and store multiple activations on the same key in a table to allow one key to be used on multiple machines.
